network/netcore/src/transport/and_then.rs

```diff
@@ -152,35 +146,28 @@ where
 {
     type Output = Result<O2, E>;
 
-    fn poll(mut self: Pin<&mut Self>, mut context: &mut Context) -> Poll<Self::Output> {
+    #[project]
+    fn poll(self: Pin<&mut Self>, mut context: &mut Context) -> Poll<Self::Output> {
+        let mut this = self.project();
         loop {
-            let (output, (f, origin)) = match self.as_mut().project().chain {
+            #[project]
```

Comment:
Is this needed because we're actually matching over the projected type?

Reply:
Yes -- Currently, all ways for referring the projected type are provided by the [`#[project]` attribute](https://docs.rs/pin-project-internal/0.4.2/pin_project_internal/attr.project.html). ([but, other ways may be provided in the future](https://github.com/taiki-e/pin-project/issues/124))